invoicer_project/static/js/index.js

```diff
@@ -39,14 +41,55 @@ document.querySelector("#log_out_button").addEventListener("click", async () =>
             },
         });
 
-        if (!response.ok) {
+        if (response.ok) {
+            localStorage.removeItem('accessToken');
+            localStorage.removeItem('refreshToken');
+            window.location.href = host + '/user/login';
+        } else if (response.status === 400) {
+            const obtainedNewTokens = await obtainNewAccessToken();
+            if (!obtainedNewTokens) {
+                localStorage.removeItem('accessToken');
+                localStorage.removeItem('refreshToken');
+                window.location.href = host + '/user/login/';
+            } else {
+                await obtainUserInitials();
+            }
+        } else {
             throw new Error(response.statusText);
         }
 
-        localStorage.removeItem('accessToken');
-        localStorage.removeItem('refreshToken');
-        window.location.href = host + '/user/login';
     } catch (error) {
         console.error(error);
+        localStorage.removeItem('accessToken');
+        localStorage.removeItem('refreshToken');
+        window.location.replace(host + '/user/login/');
     }
-});
\ No newline at end of file
+});
+
+function checkMenuItemBasedOnSection() {
+    const urlSection = new URL(window.location.href).pathname.split("/")[1];
+    const menuItems = document.querySelectorAll("#menu li");
+    menuItems.forEach((item) => {
+        item.classList.remove("active");
+    });
+    switch (urlSection) {
```

Comment:
Could we use this code instead of switch?

```suggestion
const urlSections = ['invoices', 'clients', 'items', 'statistics', 'settings'];
menuItesm[urlSections.indexOf(urlSection)].classList.add('active');
```

Reply:
Done